src/sig_stfl/lms/sig_stfl_lms_functions.c

```diff
@@ -264,3 +350,139 @@ int oqs_sig_stfl_lms_verify(const uint8_t *m, size_t mlen,
 	}
 }
 
+void oqs_secret_lms_key_free(OQS_SIG_STFL_SECRET_KEY *sk) {
+	if (sk == NULL) {
+		return;
+	}
+
+	//TODO: cleanup lock_key
+
+	if (sk->sig) {
+		OQS_MEM_insecure_free(sk->sig);
+		sk->sig = NULL;
+	}
+
+	if (sk->secret_key_data) {
+		oqs_lms_key_data *key_data = (oqs_lms_key_data *)sk->secret_key_data;
+		if (key_data) {
+			OQS_MEM_secure_free(key_data->sec_key, key_data->len_sec_key);
+			key_data->sec_key = NULL;
+
+			OQS_MEM_secure_free(key_data->aux_data, key_data->len_aux_data);
+		}
+
+		OQS_MEM_insecure_free(key_data->data);
```

Comment:
This is intended to be a variant specific deep cleaning function with knowledge of the internals.

Reply:
Let me know if you are ok with this?

Reply:
My concern was that, since the `data` field is described as user-defined, the user could (say) set `data` to point to stack memory, in which case freeing `data` would cause serious problems. Also, the `data` field is also not initialized to `NULL` by default, so freeing it could cause problems unless it's already been `malloc`ed, and it's not `malloc`ed anywhere that I can see.

That said, I'm a little shaky on what the purpose of the `data` pointer is, so I might be missing something.

Reply:
If this isn't actually a concern and/or will be addressed in a future PR, feel free to resolve this thread.

Reply:
I will just remove it until we hit/need code for a use case.